verilog/tools/kythe/indexing_facts_tree.h

```diff
@@ -0,0 +1,101 @@
+// Copyright 2017-2020 The Verible Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef VERIBLE_VERILOG_TOOLS_KYTHE_INDEXING_FACTS_TREE_H_
+#define VERIBLE_VERILOG_TOOLS_KYTHE_INDEXING_FACTS_TREE_H_
+
+#include "absl/strings/substitute.h"
+#include "common/text/token_info.h"
+#include "common/util/vector_tree.h"
+#include "verilog/tools/kythe/verilog_extractor_indexing_fact_type.h"
+
+namespace verilog {
+namespace kythe {
+
+// Anchor class represents the location and value of some token.
```

Comment:
Design-for-thought: Would it perhaps be easier if this were a `TokenInfo` and only right before emitting facts with byte offsets, pass in the `base` string to compute `.left()` and `.right()`?  Less plumbing of `base` perhaps.  See this concept in 

https://github.com/google/verible/blob/650c6cc182a182543bdd6fc6e6c63fc7b7b2f563/common/text/token_info.h#L60

https://github.com/google/verible/blob/650c6cc182a182543bdd6fc6e6c63fc7b7b2f563/common/text/tree_utils.h#L307

Reply:
I face problems like getting garbage whenever I use string_view here.
I tried to use TokenInfo in IndexingNodeData but I faced the same problems with garbage.